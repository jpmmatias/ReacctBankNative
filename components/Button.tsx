import React from 'react';
import { ImageSourcePropType } from 'react-native';
import {
	StyleSheet,
	Image,
	Dimensions,
	TouchableHighlight,
	View,
	Text,
} from 'react-native';

interface ButtonProps {
	text: string;
	textColor: string;
	textSize: number;
	icon?: ImageSourcePropType;
	backgroundColor: string;
	textColorHover?: string;
	widthSize: number;
	heightSize: number;
	textWeight:
		| 'normal'
		| 'bold'
		| '100'
		| '200'
		| '300'
		| '400'
		| '500'
		| '600'
		| '700'
		| '800'
		| '900';
	LinkTo?: string;
	handleClick?: () => void;
}

const Button = ({
	text,
	textColor,
	textSize,
	icon,
	backgroundColor,
	widthSize,
	heightSize,
	textWeight,
	LinkTo,
	handleClick,
}: ButtonProps) => {
	if (LinkTo) {
		return (
			<TouchableHighlight onPress={handleClick}>
				<View
					style={[
						{
							width: widthSize,
							backgroundColor: backgroundColor,
							height: heightSize,
							borderRadius: 20,
						},
					]}
				>
					<Text
						style={[
							{
								fontSize: textSize,
								color: textColor,
								fontWeight: textWeight,
							},
						]}
					>
						{text}
					</Text>
					{icon && <Image style={styles.icon} source={icon} />}
				</View>
			</TouchableHighlight>
		);
	}

	return (
		<TouchableHighlight
			activeOpacity={0.6}
			style={{
				width: widthSize,
				height: heightSize,
				borderRadius: 20,
			}}
			onPress={handleClick}
		>
			<View
				style={[
					{
						width: widthSize,
						backgroundColor: backgroundColor,
						height: heightSize,
						borderRadius: 20,
						justifyContent: icon ? 'space-around' : 'center',
						alignItems: 'center',
					},
				]}
			>
				<Text
					style={[
						{
							fontSize: textSize,
							color: textColor,
							fontWeight: textWeight,
						},
					]}
				>
					{text}
				</Text>
				{icon && <Image style={styles.icon} source={icon} />}
			</View>
		</TouchableHighlight>
	);
};

const styles = StyleSheet.create({
	icon: {
		height: 50,
		width: 50,
	},
	title: {
		marginTop: 5,
		fontSize: 21,
		textAlign: 'center',
		fontWeight: '500',
		color: '#1d1d1d',
		lineHeight: 24.61,
		marginBottom: 60,
	},
});

export default Button;
