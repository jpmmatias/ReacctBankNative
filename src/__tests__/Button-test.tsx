import React from 'react';
import { create } from 'react-test-renderer';
import Button from '../components/Button';

const tree = create(
	<Button
		text='Realizar DepÃ³sitos'
		handleClick={() => console.log('test')}
		textColor='#fff'
		backgroundColor='#68DE5A'
		textSize={22}
		widthSize={265.64}
		heightSize={56.97}
		textWeight='600'
	/>
);

test('button', () => {
	expect(tree).toMatchSnapshot();
});
