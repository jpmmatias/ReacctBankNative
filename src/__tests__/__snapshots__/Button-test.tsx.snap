// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 20,
      "height": 56.97,
      "width": 265.64,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#68DE5A",
          "borderRadius": 20,
          "height": 56.97,
          "justifyContent": "center",
          "width": 265.64,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "600",
          },
        ]
      }
    >
      Realizar Dep√≥sitos
    </Text>
  </View>
</View>
`;
